monitoring/monitorlib/kml.py

```diff
@@ -0,0 +1,101 @@
+# Module to parse KML file.
+
+from pykml import parser
+import re
+
+
+KML_NAMESPACE = {"kml":"http://www.opengis.net/kml/2.2"}
+
+def get_kml_root(kml_path):
+    kml = parser.parse(kml_path)
+    return kml.getroot()
+
+
+def get_folders(root):
+    return root.Document.Folder.Folder
+
+
+def get_polygon_speed(polygon_name):
+    """Returns speed unit within a polygon."""
+    result = re.search(r"\(([0-9.]+)\)", polygon_name)
+    return float(result.group(1)) if result else None
+
+def get_folder_details(folder_elem):
+    speed_polygons = {}
+    alt_polygons = {}
+    operator_location = {}
+    coordinates = ''
+    for placemark in folder_elem.xpath('.//kml:Placemark', namespaces=KML_NAMESPACE):
+        placemark_name = str(placemark.name)
+        polygons = placemark.xpath('.//kml:Polygon', namespaces=KML_NAMESPACE)
+
+        if placemark_name == 'operator_location':
+            operator_location = {
+                'lng': folder_elem.xpath(
+                    './/kml:Placemark/kml:LookAt/kml:longitude', namespaces=KML_NAMESPACE)[0],
+                'lat': folder_elem.xpath(
+                    './/kml:Placemark/kml:LookAt/kml:latitude', namespaces=KML_NAMESPACE)[0]
+            }
+        if polygons:
+            if placemark_name.startswith('alt:'):
+                polygon_coords = get_coordinates_from_kml(
+                    polygons[0].outerBoundaryIs.LinearRing.coordinates)
+                alt_polygons.update({placemark_name: polygon_coords})
+            if placemark_name.startswith('speed:'):
+                if not get_polygon_speed(placemark_name):
+                    # TODO: raise error
```

Comment:
To avoid failing silently, I think we can go ahead and actually raise the error here (e.g., `raise ValueError('Could not determine Polygon speed from Placemark "{}"'.format(placemark_name))`)